### DISCUSSION TITLE
Optional fields are defined as null

### DISCUSSION THREAD

**Discussion by SwagiWagi** [prisma/prisma#16567] [Q&A] [answered]
Hello, as stated in this section; https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#unsupported-types optional fields are defined as null values instead of undefined, why is that and can I change it to undefined?

**Comment by pantharshit00** [comment, marked as answer]
Hello @SwagiWagi,

You can read this documentation piece which goes in detail about explaining these two values in the client: https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined


Currently, it is not possible to alter this behaviour and this will also be a big breaking change for users of the library. If you are keen on proposing this change, you can open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=).

**Reply by SwagiWagi** [reply]
How would I sync these shcemas?

TypeScript:
`export interface X {
    aKey?: string;
}`

Prisma schema:
`type X {
    aKey String?
}`

**Reply by pantharshit00** [reply]
You can import types which Prisma generated: 
```ts
import { X } from '@prisma/client';
```

This type will reflect the structure which you defined in your schema. You can read this documentation piece on other types which Prisma exports: https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety#what-are-generated-types. 


 I am not sure what you mean to refer when you are saying you want to "sync these schemas"? Can you please clarify that?

**Reply by vladi-stevanovic** [reply]
👋 Hi @SwagiWagi , did you perhaps have a chance to check Harshit's question? Let us know if you still need help!

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
